### PR TITLE
Use composite

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

File: runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java [hunk 1]
```diff
@@ -34,38 +24,13 @@
 public class CompositeRegistryCreator {
     private static final Logger log = Logger.getLogger(CompositeRegistryCreator.class);
 
+    public static CompositeMeterRegistry rootRegistry;
```
Comment (ebullient) [inline runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java:27]: Can we create a static setter (or something) for this so people know to look for who calls the setter to set this value? Either that, or have the producer reference/return a static value from the Recorder ..

File: runtime/src/main/java/dev/ebullient/micrometer/runtime/MicrometerRecorder.java [hunk 2]
```diff
@@ -26,13 +32,35 @@
     private static final Logger log = Logger.getLogger(MicrometerRecorder.class);
     static final int TRIM_POS = "quarkus.micrometer.export.".length();
 
+    private volatile static CompositeMeterRegistry quarkusRegistry;
+
+    public void createRootRegistry() {
+        Clock clock = Arc.container().beanManager().createInstance().select(Clock.class).get();
+        quarkusRegistry = new CompositeMeterRegistry(clock);
+        VertxMeterBinderAdapter.meterRegistry.set(quarkusRegistry);
+        CompositeRegistryCreator.rootRegistry = quarkusRegistry;
```
Comment (ebullient) [inline runtime/src/main/java/dev/ebullient/micrometer/runtime/MicrometerRecorder.java:41]: Per comment above.. setting the CompositeRegistryCreator.rootRegistry in this way bugs me a little.

File: runtime/src/main/java/dev/ebullient/micrometer/runtime/binder/vertx/VertxMeterBinderAdapter.java [hunk 3]
```diff
@@ -19,21 +17,16 @@
 import io.vertx.core.spi.metrics.VertxMetrics;
 
 @Singleton
-public class VertxMeterBinderAdapter extends MetricsOptions implements MeterBinder, VertxMetricsFactory, VertxMetrics {
+public class VertxMeterBinderAdapter extends MetricsOptions implements VertxMetricsFactory, VertxMetrics {
```
Comment (ebullient) [inline runtime/src/main/java/dev/ebullient/micrometer/runtime/binder/vertx/VertxMeterBinderAdapter.java:20]: I think this needs to remain a MeterBinder for discovery? Either that, or something else w/in the Vert.x Binder needs to have the correct quarkus registry injected, and I'm missing that change

Comment (kenfinnigan) [reply runtime/src/main/java/dev/ebullient/micrometer/runtime/binder/vertx/VertxMeterBinderAdapter.java:20]: It's not an actual binder, so figured it wasn't needed.

The registry is set here: https://github.com/ebullient/quarkus-micrometer-extension/pull/20/files#diff-1cddb4aa56dcbcc90d6e95b9bea4aa79R40

File: runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java [hunk 4]
```diff
@@ -2,28 +2,18 @@
...
 import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
 import io.quarkus.arc.AlternativePriority;
 
 /**
  * Ensure there is one resolvable MeterRegistry.
  * <p>
- * This is invoked/created by MicrometerProcessor#createRootRegistry.
  * Fetch MeterRegistry instances using types discovered from registered Producer fields and methods.
  * If there are no instances, create a no-op/empty CompositeMeterRegistry. If there are several instances,
  * create a CompositeMeterRegistry that aggregates the collection. Otherwise, return the single discovered
```
Comment (ebullient) [inline runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java:19]: Nice to have: dump the comment?

File: runtime/src/main/java/dev/ebullient/micrometer/runtime/binder/vertx/VertxMeterBinderAdapter.java (hunk 3 above)
Comment (ebullient) [reply runtime/src/main/java/dev/ebullient/micrometer/runtime/binder/vertx/VertxMeterBinderAdapter.java:20]: Missed that. More obvious in revised version. ;)

File: runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java (hunk 4 above)
Comment (kenfinnigan) [reply runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java:19]: I removed because it's not called by that anymore

Comment (ebullient) [reply runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java:19]: No .. I mean.. remove the whole comment.. I think you've nixed all the code it explains. ;)

Comment (kenfinnigan) [reply runtime/src/main/java/dev/ebullient/micrometer/runtime/CompositeRegistryCreator.java:19]: Doh, yeah

**Pull request merged**